### PR TITLE
Parser refactor 

### LINKED ISSUE #7: Parser supports correct input format
The parser is kind of terrible. The command line format is not one used by most programmers, and it is overly strict. v0.2's parser needs to be more flexible, well designed, and support Unix style flags 

In terms of code quality, the parser doesn't have sufficient cohesion. It is actually doing three things - parsing, dispatching and validation. Ideally there should be some way of splitting it up somehow.

### PR DESCRIPTION
**Pull request by ZhangYiJiang**
Closes #7

### CODE REVIEW

File: src/main/java/seedu/todo/logic/TodoDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Selects the correct command based on the parser results
+ */
+public class TodoDispatcher implements Dispatcher {
+    public BaseCommand dispatch(ParseResult parser) throws IllegalValueException {
+        BaseCommand command;
+        
+        switch (parser.getComand()) {
+        case "add":
```
Comment (li-kai) [inline src/main/java/seedu/todo/logic/TodoDispatcher.java:16]: spaces between colons

File: src/main/java/seedu/todo/logic/arguments/Argument.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package seedu.todo.logic.arguments;
+
+import seedu.todo.commons.exceptions.IllegalValueException;
+
+abstract public class Argument<T> implements Parameter {
+    private String name;
+    private String description;
+    private String flag;
+    private boolean optional = true;
+    private boolean set = false;
```
Comment (li-kai) [inline src/main/java/seedu/todo/logic/arguments/Argument.java]: `set` in CS has many meanings. `mutate` or `setValue` would be much clearer.

**Review by li-kai — APPROVED** [review]
Stylistic changes aside, everything LGTM!

File: src/main/java/seedu/todo/logic/TodoDispatcher.java (hunk 1 above)
Comment (ZhangYiJiang) [reply src/main/java/seedu/todo/logic/TodoDispatcher.java:16]: I think this is actually correct, if https://github.com/oss-generic/process/issues/3 is anything to go by

**Pull request merged**